crates/ops-lambdas/indexer-monitor/src/handler.rs

```diff
@@ -0,0 +1,311 @@
+// Copyright (c) 2025 RISC Zero, Inc.
+//
+// All rights reserved.
+
+use std::str::FromStr;
+
+use alloy::primitives::Address;
+use anyhow::{Context, Result};
+use aws_config::Region;
+use aws_sdk_cloudwatch::types::{MetricDatum, StandardUnit};
+use aws_sdk_cloudwatch::Client as CloudWatchClient;
+use aws_smithy_types::DateTime;
+use chrono::Utc;
+use lambda_runtime::{Error, LambdaEvent};
+use serde::Deserialize;
+use std::env;
+use tracing::{debug, error, instrument};
+
+use crate::monitor::Monitor;
+
+/// Incoming message structure for the Lambda event
+#[derive(Deserialize, Debug)]
+pub struct Event {
+    pub clients: Vec<String>,
+    pub provers: Vec<String>,
+}
+
+/// Lambda function configuration read from environment variables
+struct Config {
+    db_url: String,
+    interval_seconds: i64,
+    region: String,
+    namespace: String,
+}
+
+impl Config {
+    /// Load configuration from environment variables
+    fn from_env() -> Result<Self, Error> {
+        let db_url = env::var("DB_URL")
+            .context("DB_URL environment variable is required")
+            .map_err(|e| Error::from(e.to_string()))?;
+
+        let interval_seconds = env::var("INTERVAL_SECONDS")
+            .context("INTERVAL_SECONDS environment variable is required")
+            .map_err(|e| Error::from(e.to_string()))?
+            .parse()
+            .context("INTERVAL_SECONDS must be a valid integer")
+            .map_err(|e| Error::from(e.to_string()))?;
+
+        let region = env::var("AWS_REGION").unwrap_or_else(|_| "us-west-2".to_string());
+
+        let namespace =
+            env::var("CLOUDWATCH_NAMESPACE").unwrap_or_else(|_| "indexer-monitor".to_string());
+
+        Ok(Self { db_url, interval_seconds, region, namespace })
+    }
+}
+
+/// Main Lambda handler function
+#[instrument(skip_all, err)]
+pub async fn function_handler(event: LambdaEvent<Event>) -> Result<(), Error> {
+    debug!("Lambda function started");
+
+    let config = Config::from_env()?;
+
+    let event = event.payload;
+    debug!(?event, "Received event");
+
+    let monitor = Monitor::new(&config.db_url)
+        .await
+        .context("Failed to create monitor")
+        .map_err(|e| Error::from(e.to_string()))?;
+
+    let now = Utc::now().timestamp();
+    let start_time = now - config.interval_seconds;
```

Comment:
Lambda won't guarantee we start exactly at a particular timestamp, so we can end up missing metrics here if the lambda starts late. I think we're going to need to store the timestamps we've processed up to